### PR TITLE
Exploration of the Vehicles dataset based on Startup task #2

### PR DESCRIPTION
**Pull request by opeyemiferanmi1**
Here is a summary of my exploration of the Vehicles dataset 
 
-- load the vehicle dataset from the repo
-- basic explanatory data analysis
-- split datasets into testing and training data
-- Perform algorithm check(K Nearest Neighbor, Naiye Bayes, Support Vector Classifier and Random Forest Classifier) to determine which is the best for the dataset based on accuracy score
-- train Support Vector Classifier  model from scikit-learn
-- compute the following  evaluation metric on a held-out test set
* Accuracy
* Classification report
* Confusion Matrix

  I will work on adding more documentations and Visualizations, but I will love a review of this approach.

### CODE REVIEW

File: dev/opeyemiferanmi1/helper.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    results = []
+    names = []
+    for name, model in models:
+        kfold = StratifiedKFold(n_splits=n_splits, random_state=42)
+        cv_results = cross_val_score(model, x_train, y_train, cv=kfold, scoring='accuracy')
+        results.append(cv_results)
+        names.append(name)
+        print('%s: %f (%f)' % (name, cv_results.mean(), cv_results.std()))
+
+def split(x, y, split_ratio=0.2):
```
Comment (mlopatka) [inline dev/opeyemiferanmi1/helper.py]: What is the motivation for specifying a split ratio of 0.2?
Please justify this choice in comments or int he notebook.

File: dev/opeyemiferanmi1/helper.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    """Takes in a dataframe and a list of column names to transform
+        returns a dataframe of scaled values"""
+    df_to_scale = df[column_list]
+    x = df_to_scale.values
+    scaler = StandardScaler()
+    x_scaled = scaler.fit_transform(x)
+    df_to_scale = pd.DataFrame(x_scaled, columns=df_to_scale.columns)
+    return df_to_scale
+
+def spot_check_algorithm(x_train, y_train, n_splits=10 ):
```
Comment (mlopatka) [inline dev/opeyemiferanmi1/helper.py]: What is the motivation for specifying a 10 splits?
Please justify this choice in comments or int he notebook.

**Comment by opeyemiferanmi1** [comment]
Hello  mlopatka, I discovered my PR was not properly made. I do have a
folder assigned to my username in the *dev* directory of the project. I
have also tried creating this file and merging my PR into it, but I do not
get how to do that. Do I create the file and then make a new PR (as issue
#2 has been closed) or I just continue working. Please look into this and
advise me.

**Comment by dzeber** [comment]
Hi @opeyemiferanmi1, no problem. You can just make the necessary changes, commit them to this same branch (`My-contributions`) and push. The changes will automatically show up in this PR.

Also, don't worry about issues being closed - we had some of them getting closed automatically by accident, but all issues remain open.

File: dev/opeyemiferanmi1/helper.py (hunk 1 above)
Comment (opeyemiferanmi1) [reply dev/opeyemiferanmi1/helper.py]: I have replaced the split ratio with stratified kfold to ensure each fold has the same proportion/distribution.
I have also excluded all .xml and .iml files from PRs by adding the filetypes to the project .gitignore
I also included some visualizations for better understanding of the model performance.

Comment (opeyemiferanmi1) [reply dev/opeyemiferanmi1/helper.py]: Thanks for the review

File: dev/opeyemiferanmi1/helper.py (hunk 2 above)
Comment (opeyemiferanmi1) [reply dev/opeyemiferanmi1/helper.py]: I have added this justification to the notebook

**Comment by opeyemiferanmi1** [comment]
@dzeber @mlopatka  In the process of  excluded all .xml and .iml files from PRs by adding the filetypes to the project .gitignore. I did add this file types into project .gitignore which has resulted to this conflict, I had tried removing the files I added and created a .gitignore file for my branch. But I still have this conflict unresolved, Please help look into this.

**Comment by dzeber** [comment]
Thanks for these updates! The notebook is much improved, readable and the analysis is well-explored. Also the helper module is clear and well-documented. I will fix the merge conflict.

**Comment by opeyemiferanmi1** [comment]
Thanks for the review
